extension/src/astObjects/dclTile.ts

```diff
@@ -0,0 +1,224 @@
+import { Position, Range } from 'vscode';
+import { DclAtom } from './dclAtom';
+import { DclAttribute } from './dclAttribute';
+import { IDclContainer, IDclFragment } from './dclInterfaces';
+
+
+export class DclTile implements IDclContainer {
+    
+	get line(): number {
+        return this.firstAtom.line;
+    }
+
+    get column(): number {
+        return this.firstAtom.column;
+    }
+
+    get asTile(): DclTile {
+        return this;
+    }
+
+    get asContainer(): IDclContainer {
+        return this;
+    }
+
+    get isComment(): boolean { 
+        return false; 
+    }
+
+    get isBlockComment(): boolean { 
+        return false; 
+    }
+    
+    get isString(): boolean { 
+        return false; 
+    }
+
+    get range(): Range {
+        const begin = this.firstAtom.range.start;
+        const close = this.lastAtom.range.end;
+        return new Range(begin, close);
+    }
+
+    equal(atom: IDclFragment): boolean {
+        return JSON.stringify(this) === JSON.stringify(atom);
+    }
+
+    contains(position: Position): boolean {
+        return this.range.contains(position);
+    }
+
+    getAtomFromPosition(position: Position): IDclFragment {
+        if (this.contains(position)) {
+            for (let i = 0; i < this.length; i++) {
+                if (!this.atoms[i].contains(position)) {
+                    continue;
+                }
+                return this.atoms[i].getAtomFromPosition(position);
+            }
+        }
+        return null;
+    }
+
+    public atoms: Array<IDclFragment>;
+
+    get length(): number {
+        return this.atoms?.length ?? 0;
+    }
+
+    get firstAtom(): IDclFragment {
+        return this.atoms[0].asContainer?.firstAtom ?? this.atoms[0];
+    }
+
+    get lastAtom(): IDclFragment {
+        return this.atoms[this.length - 1].asContainer?.lastAtom ?? this.atoms[this.length - 1];
+    }
+
+    get firstNonComment(): IDclFragment {
+        for (let i = 0; i < this.length; i++) {
+            const item = this.atoms[i];
+            if (item.isComment){
+                continue;
+            }
+            if (item.asContainer) {
+                return null;
```

Comment:
This rejects anything that that is a container because any **non-root** tile should always have at least ": tiletype {" as prefix DclAtoms. If this is confusing, I can add a comment.